scripting/src/main/java/org/infinispan/scripting/impl/ScriptingManagerImpl.java

```diff
@@ -0,0 +1,192 @@
+import org.infinispan.scripting.logging.Log;
+import org.infinispan.transaction.LockingMode;
+import org.infinispan.transaction.TransactionMode;
+import org.infinispan.util.concurrent.IsolationLevel;
+import org.infinispan.util.logging.LogFactory;
+
+@Scope(Scopes.GLOBAL)
+public class ScriptingManagerImpl implements ScriptingManager {
+   private static final String DEFAULT_SCRIPT_EXTENSION = "js";
+   private static final String SCRIPT_CACHE = "___script_cache";
+   private static final String COMPILED_SCRIPT_CACHE = "___compiled_script_cache";
+   private static final Log log = LogFactory.getLog(ScriptingManagerImpl.class, Log.class);
+   private EmbeddedCacheManager cacheManager;
+   private ScriptEngineManager scriptEngineManager;
+   private Cache<String, String> scriptCache;
+   private Cache<String, CompiledScript> compiledScriptCache;
+   
+
+   public ScriptingManagerImpl() {
+   }
+
+   @Inject
+   public void initialize(final EmbeddedCacheManager cacheManager) {
+      this.cacheManager = cacheManager;
+      ClassLoader classLoader = cacheManager.getCacheManagerConfiguration().classLoader();
+      this.scriptEngineManager = new ScriptEngineManager(classLoader);
+      configureCache(SCRIPT_CACHE, getScriptCacheConfiguration());
+      configureCache(COMPILED_SCRIPT_CACHE, getCompiledScriptCacheConfiguration());
+   }
+   
+   private void configureCache(String name, ConfigurationBuilder builder) {
+      Configuration cacheConfiguration = cacheManager.getCacheConfiguration(name);
+      if (cacheConfiguration == null) {
+         cacheManager.defineConfiguration(name, builder.build());
+      }
+   }
+   
+   Cache<String, String> getScriptCache() {
+      if (scriptCache == null) {
+         scriptCache = cacheManager.getCache(SCRIPT_CACHE);
+      }
+      return scriptCache;
+   }
+   
+   Cache<String, CompiledScript> getCompiledScriptCache() {
+      if (compiledScriptCache == null) {
+         compiledScriptCache = cacheManager.getCache(COMPILED_SCRIPT_CACHE);
+      }
+      return compiledScriptCache;
+   }
+   
+   private ConfigurationBuilder getScriptCacheConfiguration() {
+      GlobalConfiguration globalConfiguration = cacheManager.getGlobalComponentRegistry().getGlobalConfiguration();
+      CacheMode cacheMode = globalConfiguration.isClustered() ? CacheMode.REPL_SYNC : CacheMode.LOCAL;
+
+      ConfigurationBuilder cfg = new ConfigurationBuilder();
+      cfg.transaction()
+            .transactionMode(TransactionMode.TRANSACTIONAL).invocationBatching().enable()
+            .transaction().lockingMode(LockingMode.PESSIMISTIC).syncCommitPhase(true).syncRollbackPhase(true)
+            .locking().isolationLevel(IsolationLevel.READ_COMMITTED).useLockStriping(false)
+            .clustering().cacheMode(cacheMode).sync()
+            .stateTransfer().fetchInMemoryState(true).awaitInitialTransfer(false)
+            .compatibility().enable().marshaller(new GenericJBossMarshaller())
```

Comment:
I'd like users to be able to store scripts over HotRod/REST to the script cache, but I want to store it in java.lang.String format in the actual cache.